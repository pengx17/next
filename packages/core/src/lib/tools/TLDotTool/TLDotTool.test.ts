import { TLTestApp } from '~test'

describe('When using the tool', () => {
  it('Starts in idle', () => {
    new TLTestApp().selectTool('dot').expectToBeIn('dot.idle')
  })

  it('Transitions to creating and creates a dot shape on pointer down', () => {
    const app = new TLTestApp()
    app.deleteShapes([...app.document.shapes]).selectTool('dot')
    expect(app.shapes.size).toBe(0)
    app.pointerDown([100, 100])
    expect(app.shapes.size).toBe(1)
    const shape = app.getShapesArray()[0]
    expect({ ...shape.model, id: 'test_dot' }).toMatchSnapshot('created dot')
  })

  it('Cancels creating a shape when escape is pressed', () => {
    const app = new TLTestApp()
    app
      .deleteShapes([...app.document.shapes])
      .expectSelectedIdsToBe([])
      .selectTool('dot')
      .pointerDown([100, 100])
      .pointerMove([100, 150])
      .keyDown('Escape')
      .pointerUp()
    expect(app.isIn('dot.idle')).toBe(true)
    expect(app.shapes.size).toBe(0)
  })

  // it('Transitions from idle to select.idle on Escape', () => {
  //   new TLTestApp()
  //     .selectTool('dot')
  //     .expectToBeIn('dot.idle')
  //     .keyDown('Escape')
  //     .expectToBeIn('select.idle')
  // })
})
