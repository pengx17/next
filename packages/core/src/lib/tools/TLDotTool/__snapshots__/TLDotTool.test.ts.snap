// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using the tool Transitions to creating and creates a dot shape on pointer down: created dot 1`] = `
TLDotShape {
  "clone": [Function],
  "getBounds": [Function],
  "getCachedSerialized": [Function],
  "getCenter": [Function],
  "getRotatedBounds": [Function],
  "getSerialized": [Function],
  "hideContextBar": false,
  "hideResizeHandles": true,
  "hideRotateHandle": true,
  "hideSelection": true,
  "hideSelectionDetail": true,
  "hitTestBounds": [Function],
  "hitTestLineSegment": [Function],
  "hitTestPoint": [Function],
  "isAspectRatioLocked": false,
  "isDirty": true,
  "canEdit": false,
  "isStateful": false,
  "lastSerialized": Object {},
  "nonce": 0,
  "onHandleChange": [Function],
  "onResize": [Function],
  "props": Object {
    "id": "test_dot",
    "parentId": "page1",
    "point": Array [
      94,
      94,
    ],
    "radius": 6,
    "type": "dot",
  },
  "showCloneHandles": false,
  "canUnmount": false,
  "type": "dot",
  "validateProps": [Function],
  "version": 1,
}
`;
